test/test_hpyerr.py

```diff
@@ -200,6 +200,10 @@ def check_exception(cls):
 
     def test_h_unicode_exceptions(self):
         import pytest
+        pytest.xfail(
```

Comment:
pypy's apptest don't support xfail. Can we just skip it instead?

Reply:
What happens with the `pytest.mark.xfail` in test_importing.py?

Reply:
a completely different thing, in the best pypy tradition :man_facepalming: 
`pytest.mark.*` are handled by pytest at collection time, so xfail works properly.
However, once you are inside the test it's a different story: since it's an apptest, the `pytest` module which is imported inside the test is a dummy one which implements only `skip` and `raises`: see `pypy/tool/pytest/objspace.py:maketestobjspace()` to see where the (dark) magic happens.

So to summarize: I withdraw my suggestion above, you can just use `@pytest.mark.xfail`

Reply:
Woot. `@pytest.mark.xfail` used instead.